src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,146 @@
+    static void beforeAll() {
+        registrationService = new RegistrationServiceImpl();
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void registration_user_Ok() {
+        user = createUser(USER_LOGIN, USER_PASSWORD, USER_AGE);
+        user.setId(USER_ID);
+
+        User actualUser = registrationService.register(user);
+
+        Assertions.assertEquals(user, actualUser);
+    }
+
+    @Test
+    void register_nullUser_notOk() {
+        var exception = assertThrows(UserRegistrationException.class, () ->
+                        registrationService.register(null),
+                "UserRegistrationException expected for user = null");
+        assertEquals("User is null", exception.getMessage());
+    }
+
+    @Test
+    void register_nullLogin_notOk() {
+        user = createUser(null, USER_PASSWORD, USER_AGE);
+
+        var exception = assertThrows(UserRegistrationException.class, () ->
+                        registrationService.register(user),
+                "UserRegistrationException expected for user.login = null");
+        assertEquals("User's login is null", exception.getMessage());
+    }
+
+    @Test
+    void register_emptyLogin_notOk() {
+        user = createUser(EMPTY_LOGIN, USER_PASSWORD, USER_AGE);
+
+        var exception = assertThrows(UserRegistrationException.class, () ->
+                        registrationService.register(user),
+                "UserRegistrationException expected for empty user.login");
+        assertEquals("User's login is empty", exception.getMessage());
+    }
+
+    @Test
+    void register_nullPassword_notOk() {
+        user = createUser(USER_LOGIN, null, USER_AGE);
+
+        var exception = assertThrows(UserRegistrationException.class, () ->
+                        registrationService.register(user),
+                "UserRegistrationException expected for user.password = null");
+        assertEquals("User's password is null", exception.getMessage());
+    }
+
+    @Test
+    void register_nullAge_notOk() {
+        user = createUser(USER_LOGIN, USER_PASSWORD, null);
+
+        var exception = assertThrows(UserRegistrationException.class, () ->
+                        registrationService.register(user),
+                "UserRegistrationException expected for user.age = null");
+        assertEquals("User's age is null", exception.getMessage());
+    }
+
+    @Test
+    void register_largeAge_notOk() {
+        user = createUser(USER_LOGIN, USER_PASSWORD, LARGE_AGE);
+
+        var exception = assertThrows(UserRegistrationException.class, () ->
+                        registrationService.register(user),
+                "UserRegistrationException expected for large user.age");
+        assertEquals("User's age is too big", exception.getMessage());
+    }
+
+    @Test
+    void register_userAlreadyExist_notOk() {
+        user = createUser(USER_LOGIN, USER_PASSWORD, USER_AGE);
+        registrationService.register(user);
```

Comment:
You can’t use method `register()` here because you testing this method. You need to add a user to the base like this: `Storage.people.add()`.